### PR TITLE
Prevent native auth

### PR DESCRIPTION
**Pull request by Dreanad**
I've added a viewPath when prevent_native_auth is enabled to remove the form on signin page 

and little things ( mv fr lang in the right folder, fix providers images assets )

### CODE REVIEW

File: controllers/Handle.php [hunk 1]
```diff
@@ -26,7 +26,7 @@ class Handle extends Controller
     /**
      * Defines a collection of actions available without authentication.
      */
-    protected array $publicActions = [
+    protected $publicActions = [
```
Comment (LukeTowers) [inline controllers/Handle.php:29]: What was the reason for this change?

File: Plugin.php [hunk 2]
```diff
@@ -140,13 +140,25 @@ protected function configureProviders(): void
...
+        if(Config::get('winter.sso::prevent_native_auth'))  {
+            Event::listen('backend.page.beforeDisplay', function ($controller, $action) {
+                if(!$controller instanceof \Backend\Classes\Controller) {
+                    return;
+                }
+                if ($action == 'signin') {
+                    $controller->addViewPath(plugins_path('winter/sso/views/auth'));
+                }
+            });
+        }
```
Comment (LukeTowers) [inline Plugin.php:153]:
```suggestion
        // Extend the auth controller to add the SSO login buttons only if the native auth is disabled
        if (Config::get('winter.sso::prevent_native_auth', false))  {
            \Backend\Controllers\Auth::extend(function ($controller) {
                $controller->addViewPath(plugins_path('winter/sso/controllers/auth/prevent_native'));
            });
        }
```

Could we use the above method and move the view file to the above path to make it a bit more clear what is happening?

File: controllers/Handle.php (hunk 1 above)
Comment (Dreanad) [reply controllers/Handle.php:29]: image

This variable must be typed in the controller from which it is extended i think

File: Plugin.php (hunk 2 above)
Comment (Dreanad) [reply Plugin.php:153]: Sure, I'll do that, thanks

Comment (Dreanad) [reply Plugin.php]: Yes i tested it its working well =)

File: Plugin.php [hunk 3]
```diff
@@ -140,13 +140,20 @@ protected function configureProviders(): void
      */
     protected function extendAuthController(): void
     {
+        // Extend the auth controller to add the SSO login buttons only if the native auth is disabled
+        if (Config::get('winter.sso::prevent_native_auth', false))  {
+            \Backend\Controllers\Auth::extend(function ($controller) {
+                $controller->addViewPath(plugins_path('winter/sso/controllers/auth/prevent_native'));
```
Comment (LukeTowers) [inline Plugin.php:146]:
```suggestion
                // Disable the login form visually
                $controller->addViewPath(plugins_path('winter/sso/controllers/auth/prevent_native'));

                // Disable server processing of any auth AJAX handlers to protect against manually crafted requests
                $controller->bindEvent('ajax.beforeRunHandler', function ($handler) {
                    if ($handler === 'onSubmit') {
                        throw new \ApplicationException("Native authentication is disabled.");
                    }
                });
```

Comment (LukeTowers) [reply Plugin.php:146]: @Dreanad let me know what you think about the above suggestion

**Comment by mjauvin** [comment]
@Dreanad Are you still planning on completing this? I'm working on other changes and this PR would help.

**Comment by mjauvin** [comment]
Note: The method for implementing the `prevent_native_auth` is nice,  but it doesn't work if Winter.TailwindUI plugin is active.

**Comment by LukeTowers** [comment]
@mjauvin it should work when we add the AJAX handler blocking, and this PR could also include the appropriate view overrides for the TailwindUI plugin.

**Comment by LukeTowers** [comment]
Replaced by #7

**Pull request closed without merging**